dev/ci/ci-common.sh

```diff
@@ -12,16 +12,25 @@ ls `pwd`/bin
 mathcomp_CI_BRANCH=master
 mathcomp_CI_GITURL=https://github.com/math-comp/math-comp.git
 
+# Where we clone and build external developments
+CI_BUILD_DIR=`pwd`/_build_ci
+
 # git_checkout branch
 git_checkout()
 {
   local _BRANCH=${1}
   local _URL=${2}
   local _DEST=${3}
 
-  echo "Checking out ${_DEST}"
-  git clone --depth 1 -b ${_BRANCH} ${_URL} ${_DEST}
-  ( cd ${3} && echo "${_DEST}: `git log -1 --format='%s | %H | %cd | %aN'`" )
+  mkdir -p ${CI_BUILD_DIR}
+  cd ${CI_BUILD_DIR}
+
+  if [ ! -d ${_DEST} ] ; then
+    echo "Checking out ${_DEST}"
+    git clone --depth 1 -b ${_BRANCH} ${_URL} ${_DEST}
+  fi
+  ( cd ${_DEST} && git pull &&
```

Comment:
After thinking a bit more I am not sure I am yet happy with the current behavior if `${CI_BUILD_DIR}/${_DEST}` exists. For instance, imagine you run `make ci-blah` in some branch `b1`. Then, you checkout branch `b2` that sets some different branch for the contrib, the `git pull` won't checkout the correct branch and repository.

That is to say, `git checkout b1 repo2 CompCert` may end up really checking out branch `b2` in `repo3`.

What do you think?